examples/nami/Pkh2Pkh.purs

```diff
@@ -0,0 +1,149 @@
+import Data.Map as Map
+import Data.Maybe (Maybe(Just, Nothing), maybe)
+import Data.Newtype (unwrap)
+import Data.Tuple (fst)
+import Effect (Effect)
+import Effect.Class (liftEffect)
+import Effect.Aff (error, launchAff_)
+import Effect.Aff.Class (liftAff)
+import Effect.Console as Console
+import QueryM
+  ( QueryM
+  , defaultServerConfig
+  , getWalletAddress
+  , mkOgmiosWebSocketAff
+  , utxosAt
+  )
+import Serialization as Serialization
+import Types.ByteArray (byteArrayToHex)
+import Types.POSIXTimeRange
+  ( Extended(NegInf, PosInf)
+  , Interval(Interval)
+  , LowerBound(LowerBound)
+  , UpperBound(UpperBound)
+  )
+import Types.Transaction
+  ( NetworkId(Testnet)
+  , Transaction(Transaction)
+  , TransactionOutput(TransactionOutput)
+  , TxBody(TxBody)
+  , emptyTransactionWitnessSet
+  )
+import Types.UnbalancedTransaction (UnbalancedTx(UnbalancedTx))
+import Types.Value as Value
+import Untagged.Union (asOneOf)
+import Wallet (mkNamiWalletAff)
+
+main :: Effect Unit
+main = launchAff_ $ do
+  wallet <- Just <$> mkNamiWalletAff
+  ws <- mkOgmiosWebSocketAff "ws:127.0.0.1:1337"
+  tx <- runReaderT
+    buildTransaction
+    { ws
+    , wallet
+    , serverConfig: defaultServerConfig
+    }
+  liftEffect $
+    Console.log
+      <<< byteArrayToHex
+      <<< Serialization.toBytes
+      <<< asOneOf
+      =<< Serialization.convertTransaction tx
+
+buildTransaction :: QueryM Transaction
+buildTransaction = either (throw <<< show) pure
+  =<< balanceTxM
+  =<< buildUnbalancedTransaction
+
+buildUnbalancedTransaction :: QueryM UnbalancedTx
+buildUnbalancedTransaction = do
+  ownAddress <- mthrow "Failed to get wallet address" getWalletAddress
+  inputs <-
+    map fst
+      <<< Map.toUnfoldable
+      <<< unwrap
+      <$> mthrow "Failed to get utxos" (utxosAt ownAddress)
+  pure $ UnbalancedTx
+    { transaction: Transaction
+        { body: TxBody
+            { inputs
+            , outputs: Array.singleton $ TransactionOutput
+                { address: ownAddress
+                , amount: Value.lovelaceValueOf $ BigInt.fromInt 2000000
+                , data_hash: Nothing
+                }
+            -- ??
+            , fee: Value.mkCoin 0
+            , network_id: Just Testnet
+            , certs: Nothing
+            , collateral: Nothing
+            , auxiliary_data_hash: Nothing
+            , mint: Nothing
+            , required_signers: Nothing
+            , script_data_hash: Nothing
+            , ttl: Nothing
+            , update: Nothing
+            , validity_start_interval: Nothing
+            , withdrawals: Nothing
+            }
+        , is_valid: true
+        , witness_set: emptyTransactionWitnessSet
+        , auxiliary_data: Nothing
+        }
+    , requiredSignatories: Map.empty
+    , utxoIndex: Map.empty
+    , validityTimeRange: Interval
+        { from: LowerBound NegInf true
+        , to: UpperBound PosInf true
+        }
+    }
+
+throw :: forall (a :: Type). String -> QueryM a
```

Comment:
Entirely optional, you could make this more polymorphic via
`throw :: forall (a :: Type) (m :: Type -> Type). MonadAff m => String -> m a`
I suspect some prefer this flexibility and others prefer concrete types for readability?

Reply:
If this were to be exported or used with any other monad stack, yes, I would make it more polymorphic. I don't think there would be enough of a benefit here though